### PR TITLE
Feature/adding ideas

### PR DESCRIPTION
**Pull request by JeffKersting**
#### What does  this PR do?
-Allows users to add and delete idea cards. Minor change to grid-layout styling.
#### Where should the reviewer start?
-Main.js line 16-78. Grid layout changes at styles.css lines 6-9 and 144.
#### How should this be manually tested?
-Open index.html in any web browser (2019 or later). Add text in title and body field and click save. Click delete icon in displayed idea card.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -9,5 +9,70 @@ var searchButton = document.querySelector(".search-button");
...
+
+function deleteIdea(event){
+  var id = parseInt(event.target.parentNode.parentNode.id);
+  for(var i = 0; i < savedIdeas.length; i++){
+    if(savedIdeas[i].id === id){
+      savedIdeas.splice(i, 1)
+    }
+  showIdeas();
+  }
+};
```
Comment (JeffKersting) [inline src/main.js:78]: Add favorite function here.

File: index.html [hunk 2]
```diff
@@ -16,54 +16,18 @@ <h2>Filter Starred Ideas</h2>
...
-        </span>
-        <p class="idea-title">Idea Title</p>
-        <p class="idea-body">Idea body. Dont ever play yourself. Every chance I get, I water the plants, lion! Cloth talk! I am dying on the inside, but that's probably just gas.</p>
-        <span class="comment">
-          <img src="assets/comment.svg" class="comment-button" alt="comment button">
-          <label for="comment-button">Comment</label>
-        </span>
-      </article>
-      <article class="idea-container">
+      <article class="idea-container hidden">
```
Comment (DrewBradley) [inline index.html:30]: We can probably get rid of this hidden card because it is being created by our javascript.

File: src/idea.js [hunk 3]
```diff
@@ -15,5 +15,3 @@ class Idea {
 
   }
 }
-
-export.module = Idea;
```
Comment (DrewBradley) [inline src/idea.js:19]: This class does not need to be exported.

File: styles.css [hunk 4]
```diff
@@ -129,7 +133,6 @@ button {
   margin-top: 27em;
   margin-left: 20%;
   padding: 1%;
-  justify-content: space-between;
```
Comment (DrewBradley) [inline styles.css:132]: Dave Matthews would be so upset.

File: styles.css [hunk 5]
```diff
@@ -138,7 +141,7 @@ button {
   flex-direction: column;
   width: 30%;
   height: 325px;
-  margin: 1%;
```
Comment (DrewBradley) [inline styles.css:141]: Great job trouble shooting this and doing the math to find the exact percentage.

File: src/main.js [hunk 6]
```diff
@@ -9,5 +9,70 @@ var searchButton = document.querySelector(".search-button");
...
+function saveIdea() {
+  if (titleInput.value && bodyInput.value) {
+    var idea = new Idea(titleInput.value, bodyInput.value);
+    savedIdeas.push(idea);
+  }
+  titleInput.value = "";
+  bodyInput.value = "";
+  saveButton.disabled = true;
+  showIdeas();
+  console.log('hot damn')
```
Comment (colefiscus) [inline src/main.js:30]: I think this is a great function, but we got to remember to delete this!

Comment (DrewBradley) [reply src/main.js:30]: HOT DAMN!

File: src/main.js [hunk 7]
```diff
@@ -9,5 +9,70 @@ var searchButton = document.querySelector(".search-button");
...
+function checkEventTarget(event){
+  if(event.target.classList.contains("delete")){
+    deleteIdea(event);
+  } else if (event.target.className === "like"){
+    favoriteIdea(event);
+  }
+};
+
+function deleteIdea(event){
+  var id = parseInt(event.target.parentNode.parentNode.id);
```
Comment (colefiscus) [inline src/main.js:71]: I'm curious if parentNode or parentElement has any difference still.

**Review by colefiscus — APPROVED** [review]

**Review by DrewBradley — APPROVED** [review]

**Pull request merged**